packages/zero-client/src/client/zero.ts

```diff
@@ -1217,6 +1227,7 @@ export class Zero<
       this.#options.push,
       this.#options.maxHeaderLength,
       additionalConnectParams,
+      this.#aliveClientsManager,
```

Comment:
maybe we should assert this.#aliveClientsManager is not undefined here...

it shouldn't be possible for it to be undefined because we `await this.clientGroupID` above.  

Alternatively we could make this.#aliveClientsManager  a Promise<ActiveClientsManager> and await it here.

Reply:
I tried making the ActiveClientsMananger sync. It lead to a race condition with the "event" when closing.

In general I try to make construction sync and hide the async details in the implementation but in this case it cleaned things up to make the "constructor" async.

Reply:
What I'm hoping to address is that with the current code it appears this.#aliveClientsManager could be undefined here (and thus we wouldn't send the active clients on connect).  

I think we can change the createSocket arg from 

```
 aliveClientsManager:
    | Pick<ActiveClientsManager, 'getActiveClients'>
    | undefined,
```

to just
```
 aliveClientsManager: Pick<ActiveClientsManager, 'getActiveClients'>
 ```
 
 If we made this.#aliveClientsManager a promise that resolved to activeClientsManager when it was constructred, and then here we would just pass
 
 `await this.#aliveClientsManager`